homeassistant/components/mill/coordinator.py

```diff
@@ -40,3 +52,106 @@ def __init__(
             update_method=mill_data_connection.fetch_heater_and_sensor_data,
             update_interval=update_interval,
         )
+
+
+class MillHistoricDataUpdateCoordinator(DataUpdateCoordinator):
```

Comment:
codecov is complaining about poor coverage. Considering the code is a bit hairy, I think it would make a lot of sense to have all the paths covered.

Reply:
I have added some more testing.

Is the coverage a bit buggy?
How can line 79 be covered, but not 79-81 ?
https://app.codecov.io/gh/home-assistant/core/pull/130406/blob/homeassistant/components/mill/coordinator.py#L78

Reply:
Yes, coverage is not 100% reliable.
Run it locally instead `pytest --cov=homeassistant.components.mill --cov-report html tests/components/mill`, I've never seen that not giving reliable results.

Reply:
<img width="681" alt="image" src="https://github.com/user-attachments/assets/216cf797-b174-43a6-8e27-70d708ea5441" />

Reply:
OK, and what does that prove?

Reply:
That the coverage now is good?

Reply:
There are 4 uncovered lines though, it seems to be about handling error cases.
Can't you add some tests for that?

Reply:
<img width="770" alt="image" src="https://github.com/user-attachments/assets/bafd1221-a439-4bd9-9219-5fce77647c46" />